app/src/main/java/com/babylon/wallet/android/presentation/main/MainViewModel.kt

```diff
@@ -75,9 +75,20 @@ class MainViewModel @Inject constructor(
     private var countdownJob: Job? = null
 
     @OptIn(ExperimentalCoroutinesApi::class)
-    val observeP2PLinks = getProfileUseCase.observeIsInitialized()
-        .filter { isInitialized -> isInitialized }
-        .flatMapLatest { p2PLinksRepository.observeP2PLinks() }
+    val observeP2PLinks = getProfileUseCase.state
+        .flatMapLatest { profileState ->
+            /**
+             * Observe p2p links after the profile has been restored,
+             * otherwise skip the first value and observe only the upcoming changes.
+             * This ensures the p2p links connection is not being unnecessarily established,
+             * but is being established upon request (e.g when a new p2p link has been added),
+             * even if the profile has not yet been restored
+             */
+            when (profileState) {
+                is ProfileState.Restored -> p2PLinksRepository.observeP2PLinks()
```

Comment:
not really related to this PR, but I noticed that this keeps emitting even when there are not any new p2p links.
So should we `distinctUntilChanged` there?

Reply:
Makes sense, as it emits every time the profile changes, even if the p2p links stay the same.